.circleci/config.yml

```diff
@@ -83,6 +83,24 @@ conda-steps: &conda-steps
         command: |
           tox -e $TEST_WHEELINSTALL_ENV --recreate --installpkg dist/*-none-any.whl
 
+gallery-steps: &gallery-steps
+  steps:
+    - checkout
+    - restore_cache:
+        keys:
+          - ophys-data-cache
+    - run:
+        <<: *initialize-venv
+    - run:
+        name: Run the gallery tests
+        command: |
+          . ../venv/bin/activate
+          pip install tox
+          tox -e $TEST_TOX_ENV
+    - save_cache:
+        key: ophys-data-cache
```

Comment:
I suggest to include the revision. The problem with this approach is that the cache will never be invalidated.

Consider that the prefix `ophys-data-cache` has already been used, you could instead use `gallery-test-{{ .Revision }}`

```
- restore_cache:
    keys: gallery-test-

[...]

- save_cache:
    key: gallery-test-{{ .Revision }}
```

From  https://circleci.com/docs/2.0/caching/#clearing-cache : 

> Tip: Caches are immutable so it is useful to start all your cache keys with a version prefix, for example v1-.... This 
> enables you to regenerate all of your caches by incrementing the version in this prefix.

Reply:
The idea is the following:
* Use revision for caching. Then specifying `gallery-test-` ensure the latest available version is restored and will avoid unwanted download. See [Restoring Cache](https://circleci.com/docs/2.0/caching/#restoring-cache)
* To ensure the updated version is effectively downloaded, checksum of the current version (restored from cache) should be computed and compared with the expected one ... if not the expected one, the data would effectively be re-downloaded and cached.